app/assets/javascripts/miq_tree.js

```diff
@@ -147,88 +131,168 @@ function miqTreeScrollToNode(tree, id) {
 // OnCheck handler for the belongs to drift/compare sections tree
-function miqOnCheckSections(node, tree_name) {
-  var selectedKeys = miqTreeObject(tree_name).getChecked().map(function(n) {
-    return n.key;
-  });
+function miqOnCheckSections(key, checked, tree) {
+  var selectedKeys = getSelectedKeys(tree);
 
-  var url = ManageIQ.tree.checkUrl + '?id=' + encodeURIComponent(node.key) + '&check=' + node.state.checked;
+  var url = ManageIQ.tree.checkUrl + '?id=' + encodeURIComponent(key) + '&check=' + checked;
   miqJqueryRequest(url, {data: {all_checked: selectedKeys}});
   return true;
 }
 
-function miqOnCheckGenealogy(node, treename) {
-  var tree = miqTreeObject(treename);
-  // Map the selected nodes into an array of keys
-  var selectedKeys = tree.getChecked().map(function(item) {
-    return encodeURIComponent(item.key);
-  });
+function miqOnCheckGenealogy(key, checked, tree) {
+  var selectedKeys = getSelectedKeys(tree);
+
   // Activate toolbar items according to the selection
   miqSetToolbarCount(selectedKeys.length);
   // Inform the backend about the checkbox changes
   miqJqueryRequest(ManageIQ.tree.checkUrl + '?all_checked=' + selectedKeys, {beforeSend: true, complete: true});
 }
 
-function miqCheckAll(cb, treename) {
-  var tree = miqTreeObject(treename);
+// Services -> Catalogs -> Catalog Items -> Edit item -> Tenants tree
+function miqOnCheckTenantTree(key) {
+  sendDataWithRx({
+    controller: 'catalogItemFormController',
+    key: key,
+  });
+}
+
+/**
+ * Returns the redux tree by its name if it is in the redux store.
+ * @param  {string} treeName The tree name in the redux store.
+ * @return {object}          Tree object.
+ */
+function getReduxTree(treeName) {
+  return ManageIQ.redux.store.getState()[treeName];
+}
+
+function miqCheckAll(cb, treeName) {
   // Set the checkboxes according to the master checkbox
-  if (cb.checked) {
-    tree.checkAll({silent: true});
-  } else {
-    tree.uncheckAll({silent: true});
-  }
+  ManageIQ.redux.store.dispatch({namespace: treeName, type: '@@tree/checkAll', value: cb.checked});
+
   // Map the selected nodes into an array of keys
-  var selectedKeys = tree.getChecked().map(function(item) {
-    return encodeURIComponent(item.key);
-  });
+  var selectedKeys = [];
+  var tree = getReduxTree(treeName);
+
+  for (var property in tree) {
+    if (tree.hasOwnProperty(property) && property !== '') {
+      selectedKeys.push(encodeURIComponent(tree[property].attr.key));
+    }
+  }
+
   // Activate toolbar items according to the selection
   miqSetToolbarCount(selectedKeys.length);
   // Inform the backend about the checkbox changes
   miqJqueryRequest(ManageIQ.tree.checkUrl + '?check_all=' + encodeURIComponent(cb.checked) + '&all_checked=' + selectedKeys);
 }
 
+// /////////////////////////////////////////////////////////////////////////////
+// END: ONCHECK FUNCTIONS
+// START: ONCLICK FUNCTIONS
+// /////////////////////////////////////////////////////////////////////////////
```

Comment:
Seriously? :rofl:

Reply:
It was all over the place 😃 every `onSomething` function scattered trough the file like they were randomized by name.

Reply:
Can you please clean this up?